CRM/Event/ICalendar.php

```diff
@@ -134,4 +124,25 @@ protected static function gCalRedirect(array $events) {
     CRM_Utils_System::redirect($url);
   }
 
+  protected static function trimEventDetails(array $event) {
+    if (!empty($event)) {
+      $event_details = $event['description'];
+
+      // Add space after paragraph
+      $event_details = str_replace('</p>', '</p> ', $event_details);
+      $event_details = strip_tags($event_details);
+
+      // Truncate Event Description and add permalink if greater than 996 characters
+      if (strlen($event_details) > 996) {
+        if (preg_match('/^.{0,996}(?=\s|$_)/', $event_details, $m)) {
+          $event_details = $m[0] . '...';
+        }
+      }
+
+      $event_details .= "\n\n<a href=\"{$event['url']}\">" . ts('View %1 Details', [1 => $event['event_type']]) . '</a>';
+      $event['description'] = $event_description;
+ 
+      return $event;
+    }
```

Comment:
This function should return the $event array unconditionally.

```suggestion
    }
    return $event;
```

Although it might be simpler for it to return nothing and pass the event by referece.

Reply:
@colemanw You're right that was super sloppy of me. I had thought about passing a parameter and then didn't do it. Anyway I changed it back to that mechanism so this should just work all the time.